### PR TITLE
Enable imds endpoint

### PR DESCRIPTION
**Pull request by gavinfish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Jenkins.getInstance().clouds.add(myCloud)
 [Azure Container Instances](https://docs.microsoft.com/en-us/azure/container-instances/) offers the fastest and simplest way to run a container in Azure, without having to provision any virtual machines and without having to adopt a higher-level service.
 
 ## Pre-requirements
-* Resource Group in West US, East US or West Europe. Get [region availability details](https://azure.microsoft.com/en-us/regions/services/).
+* Resource Group in available regions. Get [region availability details](https://azure.microsoft.com/en-us/regions/services/).
```
Comment (yuwzho) [inline README.md]: available for ACI? so the link should be with query: `https://azure.microsoft.com/en-us/global-infrastructure/services/?products=container-instances`

File: src/main/java/com/microsoft/jenkins/containeragents/util/AzureContainerUtils.java [hunk 2]
```diff
@@ -93,40 +88,28 @@ public static ListBoxModel listResourceGroupItems(String credentialsId) throws I
...
-            try {
-                return getAzureClient(credential.getMsiPort());
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        } else {
-            AzureCredentials.ServicePrincipal servicePrincipal = AzureCredentials.getServicePrincipal(credentialsId);
-            return getAzureClient(servicePrincipal);
-        }
+        TokenCredentialData token = getToken(credentialsId);
```
Comment (yuwzho) [inline src/main/java/com/microsoft/jenkins/containeragents/util/AzureContainerUtils.java]: Handle empty case? E.G. 1. Select the credential then un-select to `--- Select Azure Credentials ---`

File: src/main/resources/com/microsoft/jenkins/containeragents/aci/AciCloud/help-resourceGroup.html [hunk 3]
```diff
@@ -1,3 +1,3 @@
 <div>
-    Specify the resource group. ACI is only supported in West US, East US or West Europe.
+    Specify the resource group.
```
Comment (yuwzho) [inline src/main/resources/com/microsoft/jenkins/containeragents/aci/AciCloud/help-resourceGroup.html]: Also add the [Products available by region](https://azure.microsoft.com/en-us/global-infrastructure/services/?products=container-instances)

File: README.md (hunk 1 above)
Comment (gavinfish) [reply README.md]: done

File: src/main/java/com/microsoft/jenkins/containeragents/util/AzureContainerUtils.java (hunk 2 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/containeragents/util/AzureContainerUtils.java]: done

File: src/main/resources/com/microsoft/jenkins/containeragents/aci/AciCloud/help-resourceGroup.html (hunk 3 above)
Comment (gavinfish) [reply src/main/resources/com/microsoft/jenkins/containeragents/aci/AciCloud/help-resourceGroup.html]: done

**Pull request merged**